src/main/java/io/ticktok/client/server/rest/RestClockActions.java

```diff
@@ -38,8 +42,14 @@ private Clock getClockBy(ClockRequest clockRequest) throws IOException {
         final HttpGet httpGet = new HttpGet(urlResolver.queryParam("name", clockRequest.getName()).queryParam("schedule", clockRequest.getSchedule()).resolve());
         HttpResponse httpResponse = httpClient.execute(httpGet);
         String entity = extractEntityFrom(httpResponse);
-        new RestResponseValidator(httpResponse).validate(200, new ClockNotFoundException(entity));
-        return clockFrom(entity);
+        final List<Clock> clocks = new Gson().fromJson(entity, new TypeToken<List<Clock>>() {}.getType());
```

Comment:
are there other types of clocks ? of is there only one type == class? if there is im not sure you need to use typeToken

Reply:
You always need a typetoken for lists

Reply:
you are right....

Reply:
since you have a `return` statement here, i would refactor name to something like this:
`return topClockFrom` or `return firstClockFrom`

Reply:
in params i would pass the name and the schedule as is and not the whole `clockRequest` object, since you only need it for exception...

Reply:
done